File: ApiSuvesaPos/Datos/Class/TomaGeneral.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Datos.Models;
using Microsoft.EntityFrameworkCore;

namespace Datos.Class
{
	public class TomaGeneral
	{
		private SeePOSContext entities;

		public TomaGeneral()
		{
			entities = new SeePOSContext();
		}

		public int Crear(Models.TomaGeneral tomageneral) // registro de TomaGeneral
		{
			try
			{
				entities.TomaGenerals.Add(tomageneral);
				return entities.SaveChanges();
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public List<Models.TomaGeneralDetalle> BuscarDetalle(decimal Id)  //consultar TomaGeneral
		{
			try
			{
				List<Models.TomaGeneralDetalle> result;
					var temp = from c in entities.TomaGeneralDetalles
							   where c.IdToma == Id
							   select c;
					result = temp.ToList<Models.TomaGeneralDetalle>();
				
				if (result.Count > 0)
				{
					return result;
				}
				else
				{
					return result = null;
				}
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public List<Models.TomaGeneral> Buscar(bool porNombre, string filtro)  //consultar TomaGeneral
		{
			try
			{
				List<Models.TomaGeneral> result;
				if (porNombre == true)
				{
					var temp = from c in entities.TomaGenerals
							   select c;
					result = temp.ToList<Models.TomaGeneral>();
				}
				else
				{
					var temp = from c in entities.TomaGenerals
							   select c;
					result = temp.ToList<Models.TomaGeneral>();
				}
				if (result.Count > 0)
				{
					return result;
				}
				else
				{
					return result = null;
				}
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public int Editar(decimal id, Models.TomaGeneral nuevo) //editar TomaGeneral
		{
			try
			{
				var p = entities.TomaGenerals.Find(id);
				Models.TomaGeneral viejo = p;
				if (viejo != null)
				{
					viejo.IdToma = nuevo.IdToma;
					viejo.Fecha = nuevo.Fecha;
					viejo.Anulado = nuevo.Anulado;
					viejo.Ajustado = nuevo.Ajustado;
					viejo.NumeroAjuste = nuevo.NumeroAjuste;

					entities.Entry(viejo).State = EntityState.Modified;

					return entities.SaveChanges();
				}
				else
				{
					return 0;// no se encotro el registro solicitado.
				}
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public int Borrar(decimal id) //borrar TomaGeneral
		{
			try
			{
				var p = entities.TomaGenerals.Find(id);
				entities.Remove(p);
				return entities.SaveChanges();
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}
	}
}

File: ApiSuvesaPos/Negocio/Logica/TomaGeneral.vb
Imports Negocio.Utilidades
Namespace Logica
    Public Class TomaGeneral

        Private db As Datos.Class.TomaGeneral

        Sub New()
            Me.db = New Datos.Class.TomaGeneral
        End Sub

        Public Function Buscar(porNombre As Boolean, Filtro As String) As List(Of Datos.Models.TomaGeneral)
            Dim datos As New List(Of Datos.Models.TomaGeneral)
            datos = Me.db.Buscar(porNombre, Filtro)

            Dim Resultado As New List(Of Datos.Models.TomaGeneral)
            For Each toma As Datos.Models.TomaGeneral In datos
                Dim TomaGeneral As New Datos.Models.TomaGeneral
                TomaGeneral.IdToma = toma.IdToma
                TomaGeneral.Fecha = toma.Fecha
                TomaGeneral.Anulado = toma.Anulado
                TomaGeneral.Ajustado = toma.Ajustado
                TomaGeneral.NumeroAjuste = toma.NumeroAjuste

                For Each det As Datos.Models.TomaGeneralDetalle In Me.db.BuscarDetalle(TomaGeneral.IdToma)
                    Dim Detalle As New Datos.Models.TomaGeneralDetalle
                    Detalle.IdTomaDetalle = det.IdTomaDetalle
                    Detalle.IdToma = det.IdToma
                    Detalle.Codigo = det.Codigo
                    Detalle.CodArticulo = det.CodArticulo
                    Detalle.Barras = det.Barras
                    Detalle.Descripcion = det.Descripcion
                    Detalle.Costo = det.Costo
                    Detalle.Existencia = det.Existencia
                    Detalle.Toma = det.Toma
                    Detalle.Diferencia = det.Diferencia
                    Detalle.Contado = det.Contado

                    TomaGeneral.TomaGeneralDetalles.Add(Detalle)
                Next

                Resultado.Add(TomaGeneral)
            Next

            Return Resultado
        End Function

        Public Function Crear(Toma As Datos.Models.TomaGeneral) As String
            Return Me.db.Crear(Toma)
        End Function

        Public Function Editar(id As Long, Toma As Datos.Models.TomaGeneral) As String
            Dim res As String = Me.db.Editar(id, Toma)
            If res = "0" Then
                Return "No existe el valor"
            Else
                Return res
            End If
        End Function

        Public Function Eliminar(id As Long) As String
            Dim res As String = Me.db.Borrar(id)
            If res = "0" Then
                Return "No existe el valor"
            Else
                Return res
            End If
        End Function

    End Class
End Namespace

